packages/eslint-plugin/docs/rules/no-base-to-string.mdx

```diff
@@ -34,6 +34,9 @@ value + '';
 String({});
 ({}).toString();
 ({}).toLocaleString();
+
+// Stringifying objects or instances in an array with the `Array.prototype.join`.
```

Comment:
Technically, `class MyClass {}` is not an instance - it's a constructor - and, in fact (surprisingly?), it has a perfectly reasonable stringification as `class MyClass {}` (I guess this is because it's syntactic sugar for a `Function`).

We'd better instantiate it with `new` if you want to use it in the way the comment describes.